bgpd/bgp_damp.c

```diff
@@ -671,7 +666,52 @@ int bgp_show_dampening_parameters(struct vty *vty, afi_t afi, safi_t safi)
 		vty_out(vty, "\n");
 	} else
 		vty_out(vty, "dampening not enabled for %s\n",
-			afi == AFI_IP ? "IPv4" : "IPv6");
+			get_afi_safi_str(afi, safi, false));
+
+	return CMD_SUCCESS;
+}
+
+int bgp_show_dampening_parameters(struct vty *vty, afi_t afi, safi_t safi,
+				  uint8_t show_flags)
+{
+	struct bgp *bgp;
+	bgp = bgp_get_default();
+
+	if (bgp == NULL) {
+		vty_out(vty, "No BGP process is configured\n");
+		return CMD_WARNING;
+	}
+
+	if (!CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_ALL))
+		return bgp_print_dampening_parameters(bgp, vty, afi, safi);
+
+	if (CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP)
+	    || CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP6)) {
+		afi = CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP) ? AFI_IP
+								  : AFI_IP6;
+		FOREACH_SAFI(safi) {
+			if (strmatch(get_afi_safi_str(afi, safi, true),
+				     "Unknown"))
+				continue;
 
+			if (!CHECK_FLAG(show_flags, BGP_SHOW_OPT_JSON))
+				vty_out(vty, "\nFor address family: %s\n\n",
+					get_afi_safi_str(afi, safi, false));
+
+			bgp_print_dampening_parameters(bgp, vty, afi, safi);
```

Comment:
can you show how JSON output looks like for _all_?

Reply:
Actually there is no json option for dampening parameters, 
just added the check here as well in case if we support later.

Reply:
router# show bgp all cidr-only json 
{
 "vrfId": 0,
 "vrfName": "default",
 "tableVersion": 6,
 "routerId": "1.1.1.1",
 "defaultLocPrf": 100,
 "localAS": 1,
 "routes": { "1.1.1.1/32": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"external",
    "prefix":"1.1.1.1",
    "prefixLen":32,
    "network":"1.1.1.1\/32",
    "metric":0,
    "weight":32768,
    "peerId":"(unspec)",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"0.0.0.0",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  }
],"2.2.2.2/32": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"internal",
    "prefix":"2.2.2.2",
    "prefixLen":32,
    "network":"2.2.2.2\/32",
    "metric":0,
    "locPrf":100,
    "weight":0,
    "peerId":"192.168.56.152",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"192.168.56.152",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  }
],"10.0.2.0/24": [
  {
    "valid":true,
    "pathFrom":"internal",
    "prefix":"10.0.2.0",
    "prefixLen":24,
    "network":"10.0.2.0\/24",
    "metric":0,
    "locPrf":100,
    "weight":0,
    "peerId":"192.168.56.152",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"192.168.56.152",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  },
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"external",
    "prefix":"10.0.2.0",
    "prefixLen":24,
    "network":"10.0.2.0\/24",
    "metric":0,
    "weight":32768,
    "peerId":"(unspec)",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"0.0.0.0",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  }
],"192.168.123.245/32": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"internal",
    "prefix":"192.168.123.245",
    "prefixLen":32,
    "network":"192.168.123.245\/32",
    "metric":0,
    "locPrf":100,
    "weight":0,
    "peerId":"192.168.56.152",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"192.168.56.152",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  }
],"192.168.223.245/32": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"internal",
    "prefix":"192.168.223.245",
    "prefixLen":32,
    "network":"192.168.223.245\/32",
    "metric":0,
    "locPrf":100,
    "weight":0,
    "peerId":"192.168.56.152",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"192.168.56.152",
        "hostname":"router",
        "afi":"ipv4",
        "used":true
      }
    ]
  }
] }  } 
{
 "vrfId": 0,
 "vrfName": "default",
 "tableVersion": 3,
 "routerId": "1.1.1.1",
 "defaultLocPrf": 100,
 "localAS": 1,
 "routes": { "2001:db8::1/128": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"external",
    "prefix":"2001:db8::1",
    "prefixLen":128,
    "network":"2001:db8::1\/128",
    "metric":0,
    "weight":32768,
    "peerId":"(unspec)",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"::",
        "hostname":"router",
        "afi":"ipv6",
        "scope":"global",
        "used":true
      }
    ]
  }
],"2001:db8::2/128": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"internal",
    "prefix":"2001:db8::2",
    "prefixLen":128,
    "network":"2001:db8::2\/128",
    "metric":0,
    "locPrf":100,
    "weight":0,
    "peerId":"192.168.56.152",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"fe80::a00:27ff:fefc:2aa",
        "hostname":"router",
        "afi":"ipv6",
        "scope":"global",
        "used":true
      },
      {
        "ip":"fe80::a00:27ff:fefc:2aa",
        "hostname":"router",
        "afi":"ipv6",
        "scope":"link-local"
      }
    ]
  }
],"2001:db8:85a3::8a2e:370:7334/128": [
  {
    "valid":true,
    "bestpath":true,
    "pathFrom":"external",
    "prefix":"2001:db8:85a3::8a2e:370:7334",
    "prefixLen":128,
    "network":"2001:db8:85a3::8a2e:370:7334\/128",
    "metric":0,
    "weight":32768,
    "peerId":"(unspec)",
    "path":"",
    "origin":"incomplete",
    "nexthops":[
      {
        "ip":"::",
        "hostname":"router",
        "afi":"ipv6",
        "scope":"global",
        "used":true
      }
    ]
  }
] }  } 
router#

Reply:
also updated "show bgp all json" in commit message